upf/flowtable.c

```diff
@@ -168,12 +168,17 @@ expire_single_flow (flowtable_main_t * fm, flowtable_main_per_cpu_t * fmt,
     }
   else
     {
+      upf_main_t *gtm = &upf_main;
       upf_debug ("Flow Remove %p", f);
       pool_put (fmt->timers, e);
 
       /* hashtable unlink */
       flowtable_entry_remove (fmt, f);
 
+      vlib_decrement_simple_counter (&gtm->upf_simple_counters
+				     [UPF_FLOW_COUNTER],
+				     0, fmt - fm->per_cpu, 1);
```

Comment:
this and the other places where the FLOW_COUNTER is used are wrong. You mixed up the `thread_index` with the `index`

```suggestion
      vlib_decrement_simple_counter (&gtm->upf_simple_counters
				     [UPF_FLOW_COUNTER],
				     fmt - fm->per_cpu, 0, 1);
```

@ivan4th what do think about the way that is used to calculate the `thread_index`? `fmt - fm->per_cpu` is correct, but also quite strange.
I would rather prefer to use `vlib_get_thread_index ()` instead.

Reply:
I agree that `vlib_get_thread_index ()` is better there